src/java/org/apache/cassandra/service/accord/api/AccordRoutingKey.java

```diff
@@ -318,6 +319,119 @@ public long serializedSize(TokenKey key, int version)
         }
     }
 
+    // Allows the creation of a Range that is begin inclusive or end exclusive for a given Token
+    public static final class MinTokenKey extends TokenKey
+    {
+        private static final long EMPTY_SIZE;
+
+        @Override
+        public Range asRange()
+        {
+            AccordRoutingKey before = token.isMinimum()
+                                      ? new SentinelKey(table, true)
+                                      : new MinTokenKey(table, token.decreaseSlightly());
```

Comment:
shouldn't this be a `TokenKey`?  as start is exclusive?

Reply:
Good catch, this is pretty strange,  and `token.decreaseSlightly` isn't even implemented by all partitioners I think. Converting one of these to a `Range` would also behave oddly I think just because it would exclude everything except a `MinTokenKey` which doesn't map to any real token.

Reply:
> isn't even implemented by all partitioners I think

correct, Accord (not documented) only supports very specific partitioners... we try to test multiple when it comes to serialization but stuff like this drastically limit what can be used for the whole

Reply:
I wonder if the `asRange` for `TokenKey` and `SentinelKey` should be updated to use `MinTokenKey` for the `before` instead of `token.decreaseSlighty`?

Reply:
honestly I feel that decrease/increase should prob go away with a change to ranges to allow swapping inclusive/exclusive... I wouldn't want to do that in this patch as a ton of that lives in accord-core... I would prob tackle that as "handle all partitioners" patch or something...

In this patch I would keep using decrease/increase as its consistent.

Reply:
I talked about supporting mixed inclusivity/exclusivity ranges with Benedict and he wasn't in favor of it because he thought it would a major change and the conclusion was to use a `MinTokenKey` similar to how `Token` can be converted to a `KeyBound` in Cassandra.